clang/labs/lab1/kurdina_julia/AddWarning.cpp

```diff
@@ -0,0 +1,75 @@
+#include "clang/AST/ASTConsumer.h"
+#include "clang/AST/RecursiveASTVisitor.h"
+#include "clang/Frontend/CompilerInstance.h"
+#include "clang/Frontend/FrontendPluginRegistry.h"
+
+using namespace clang;
+
+class AddWarningConsumer : public ASTConsumer {
+  CompilerInstance &Instance;
+  bool withoutClass;
+
+public:
+  AddWarningConsumer(CompilerInstance &Instance, bool withoutClass)
+      : Instance(Instance), withoutClass(withoutClass) {}
+
+  void HandleTranslationUnit(ASTContext &context) override {
+
+    struct Visitor : public RecursiveASTVisitor<Visitor> {
+      ASTContext *context;
+      bool withoutClass;
+      Visitor(ASTContext *context, bool withoutClass)
+          : context(context), withoutClass(withoutClass) {}
+
+      bool VisitFunctionDecl(FunctionDecl *FD) {
+        if (!withoutClass) {
+          std::string name = FD->getNameInfo().getAsString();
+          if (name.find("deprecated") != std::string::npos) {
+            DiagnosticsEngine &diag = context->getDiagnostics();
+            unsigned diagID =
+                diag.getCustomDiagID(DiagnosticsEngine::Warning,
+                                     "Deprecated is contain in function name");
+            SourceLocation location = FD->getLocation();
+            diag.Report(location, diagID);
+          }
+        } else {
+          if (!FD->isCXXClassMember()) {
+            std::string name = FD->getNameInfo().getAsString();
+            if (name.find("deprecated") != std::string::npos) {
+              DiagnosticsEngine &diag = context->getDiagnostics();
+              unsigned diagID = diag.getCustomDiagID(
+                  DiagnosticsEngine::Warning,
+                  "Deprecated is contain in function name");
+              SourceLocation location = FD->getLocation();
+              diag.Report(location, diagID);
```

Comment:
This code  is duplicated in both branches of `if` statement. Could you please join checks?